From 2daae89666ad253281bb3d6a027c00a702c02eff Mon Sep 17 00:00:00 2001
Message-Id: <2daae89666ad253281bb3d6a027c00a702c02eff.1648734640.git.geliang.tang@suse.com>
From: Wan Jiabing <wanjiabing@vivo.com>
Date: Thu, 1 Apr 2021 14:46:37 +0800
Subject: [PATCH] bpf, cgroup: Delete repeated struct bpf_prog declaration
Git-commit: 2daae89666ad253281bb3d6a027c00a702c02eff
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


struct bpf_prog is declared twice. There is one declaration which is
independent on the macro at 18th line. So the below one is not needed
though. Remove the duplicate.

Signed-off-by: Wan Jiabing <wanjiabing@vivo.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Song Liu <songliubraving@fb.com>
Link: https://lore.kernel.org/bpf/20210401064637.993327-1-wanjiabing@vivo.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 include/linux/bpf-cgroup.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/linux/bpf-cgroup.h b/include/linux/bpf-cgroup.h
index 6a29fe11485d..8b77d08d4b47 100644
--- a/include/linux/bpf-cgroup.h
+++ b/include/linux/bpf-cgroup.h
@@ -458,7 +458,6 @@ int cgroup_bpf_prog_query(const union bpf_attr *attr,
 			  union bpf_attr __user *uattr);
 #else
 
-struct bpf_prog;
 struct cgroup_bpf {};
 static inline int cgroup_bpf_inherit(struct cgroup *cgrp) { return 0; }
 static inline void cgroup_bpf_offline(struct cgroup *cgrp) {}
-- 
2.34.1

